Python/optimizer.c

```diff
@@ -732,6 +754,11 @@ translate_bytecode_to_trace(
         instr++;
         // Add cache size for opcode
         instr += _PyOpcode_Caches[_PyOpcode_Deopt[opcode]];
+        /* Stop if we have reached the start */
+        if (instr == initial_instr) {
+            RESERVE(1);
+            ADD_TO_TRACE(_JUMP_TO_TOP, 0, 0, 0);
```

Comment:
Shouldn't you also add `goto done` here? As it is, it seems to emit a few more unreachable uops before it hits the `JUMP_BACKWARD` (but there could be other things once we insert executors in other places?). ISTM that `_JUMP_TO_TOP` should *always* end uop generator.